themes/default/content/blog/converting-bicep-to-pulumi/index.md

```diff
@@ -0,0 +1,357 @@
+public class App {
+    public static void main(String[] args) {
+        Pulumi.run(App::stack);
+    }
+
+    public static void stack(Context ctx) {
+        final var config = ctx.config();
+        final var resourceGroupName = config.get("resourceGroupName");
+        final var currentResourceGroup = ResourcesFunctions.getResourceGroup(GetResourceGroupArgs.builder()
+            .resourceGroupName(resourceGroupName)
+            .build());
+
+        var storage = new StorageAccount("storage", StorageAccountArgs.builder()
+            .accountName("storageaccount")
+            .kind("StorageV2")
+            .location(currentResourceGroup.applyValue(getResourceGroupResult -> getResourceGroupResult.location()))
+            .resourceGroupName(currentResourceGroup.applyValue(getResourceGroupResult -> getResourceGroupResult.name()))
+            .sku(SkuArgs.builder()
+                .name("Standard_LRS")
+                .build())
+            .build());
+
+    }
+}
+```
+
+{{% /choosable %}}
+{{% choosable language yaml %}}
+
+```yaml
+configuration:
+  resourceGroupName:
+    type: string
+resources:
+  storageaccount:
+    type: azure-native:storage:StorageAccount
+    properties:
+      accountName: storageaccount
+      kind: StorageV2
+      location: ${currentResourceGroup.location}
+      resourceGroupName: ${currentResourceGroup.name}
+      sku:
+        name: Standard_LRS
+variables:
+  currentResourceGroup:
+    fn::invoke:
+      Function: azure-native:resources:getResourceGroup
+      Arguments:
+        resourceGroupName: ${resourceGroupName}
+```
+
+{{% /choosable %}}
+{{< /chooser >}}
+
+Notice how the in the original Bicep code, we reference the location of the _implicit_ resource group via `resourceGroup().location`. When the converter generates code for the target language, it replaces this _implicit_ resource group with an explicit one by parameterizing the program by a resource group name and using that resource group when deploying the resources.
+
+## Early days for the converter
+
+Currently the converter supports most of the features of Bicep such as parameters, variables, modules, resources, outputs and a number of built-in functions. That said, the converter plugin is still in its early days and it's not yet tested against many real-world Bicep programs. I am actively working on improving it and making it support most of the Bicep language features. The source code is available on this [GitHub repository](https://github.com/Zaid-Ajaj/pulumi-converter-bicep). If you find any issues or have any feedback, please open an issue on the repository.
+
+## How does it work?
+
+The converter plugin is a native binary built using dotnet in [F#](https://dotnet.microsoft.com/en-us/languages/fsharp) and new Pulumi Converter SDK. Using the core [Pulumi](https://www.nuget.org/packages/Pulumi) nuget package, we shipped an API that allows you to easily build converter plugins that can extend the functionality of `pulumi convert`.
```

Comment:
Nit: Let's replace all uses of "dotnet" with ".NET" throughout. It's not usually referred to "dotnet" unless you're specifically referring to the CLI.

- https://github.com/dotnet/runtime#readme
- https://dotnet.microsoft.com/en-us/

```suggestion
The converter plugin is a native binary built using dotnet in [F#](https://dotnet.microsoft.com/en-us/languages/fsharp) and new Pulumi Converter SDK. Using the core [Pulumi](https://www.nuget.org/packages/Pulumi) nuget package, we shipped an API that allows you to easily build converter plugins that can extend the functionality of `pulumi convert`.
```